lib/modules/datasource/github-releases/cache/cache-base.ts

```diff
@@ -365,4 +377,25 @@ export abstract class AbstractGithubDatasourceCache<
 
     return result;
   }
+
+  public isChangelogReleaseFresh(
+    changelogRelease: ChangeLogRelease | undefined,
+    now: DateTime,
+    updateDuration: DurationLike,
+    cacheItems: Record<string, StoredItem>
+  ): boolean {
+    if (changelogRelease) {
+      const changelogReleaseTime = DateTime.fromISO(
+        changelogRelease.date.toString()
+      );
+      if (
+        !cacheItems[changelogRelease.version] &&
+        changelogReleaseTime.plus(updateDuration) >= now
+      ) {
+        return true;
+      }
+    }
```

Comment:
I would like to return earlier and make the flow more linear, but it goes another branch in the caller which is hard to test.

Reply:
🤔  It's a public function, so you could test this directly

Reply:
first if is easy, just pass null as changelog release